lib/tbd/psi.rb

```diff
@@ -1856,44 +1938,144 @@ def process(model = nil, argh = {})
       edge[:set] = json[:io][:building][:psi]
     end
 
-    # A priori, TBD applies (default) :building PSI types and values to
-    # individual edges. If a TBD JSON input file holds custom PSI sets for:
-    #   :stories
-    #   :spacetypes
-    #   :surfaces
-    #   :edges
-    # ... that may apply to individual edges, then the default :building PSI
-    # types and/or values are overridden, as follows:
-    #   custom :stories    PSI sets trump :building PSI sets
-    #   custom :spacetypes PSI sets trump aforementioned PSI sets
-    #   custom :spaces     PSI sets trump aforementioned PSI sets
-    #   custom :surfaces   PSI sets trump aforementioned PSI sets
-    #   custom :edges      PSI sets trump aforementioned PSI sets
     if json[:io]
-      if json[:io].key?(:subsurfaces) # reset subsurface U-factors (if on file)
+      # Reset subsurface U-factors (if on file).
+      if json[:io].key?(:subsurfaces)
         json[:io][:subsurfaces].each do |sub|
+          match = false
           next unless sub.key?(:id)
           next unless sub.key?(:usi)
-          match = false
 
           tbd[:surfaces].values.each do |surface|
             break if match
 
             [:windows, :doors, :skylights].each do |types|
-              if surface.key?(types)
-                surface[types].each do |id, opening|
-                  break    if match
-                  next unless opening.key?(:u)
+              break    if match
+              next unless surface.key?(types)
 
-                  match = true            if sub[:id] == id
-                  opening[:u] = sub[:usi] if sub[:id] == id
-                end
+              surface[types].each do |id, opening|
+                break    if match
+                next unless opening.key?(:u)
+                next unless sub[:id] == id
+
+                opening[:u] = sub[:usi]
+                match       = true
               end
             end
           end
         end
       end
 
+      # Reset wall-to-roof intersection type (if on file) - groups.
+      [:stories, :spacetypes, :spaces].each do |groups|
```

Comment:
Wall-to-roof intersection edge types (**:parapet** vs **:roof**) can be reset for individual stories, spacetypes or spaces. Needs a few RSpecs.